test/zebra/subscriptions_test.clj

```diff
@@ -0,0 +1,56 @@
+(ns zebra.subscriptions-test
+  (:require
+    [clojure.test :refer :all]
+    [zebra.customers :as customers]
+    [zebra.payment-methods :as payment-methods]
+    [zebra.prices :as prices]
+    [zebra.products :as products]
+    [zebra.subscriptions :as subscriptions]
+    [zebra.helpers.constants :refer [api-key]]))
+
+(deftest create-subscription
+  (let [customer (customers/create api-key)
+        customer-id (:id customer)
+        payment-method (payment-methods/create {:type "card"
+                                                :card {:number    "4242424242424242"
+                                                       :exp_month "7"
+                                                       :exp_year  "2026"
+                                                       :cvc       "314"}} api-key)
+        _attached-payment-method (customers/attach-payment-method customer-id (:id payment-method) api-key)
+        product (products/create {:name (str "test_product_" (random-uuid))} api-key)
+        product-id (:id product)
+        price (prices/create {:unit_amount 999
+                              :currency    "gbp"
+                              :recurring   {:interval "day"}
+                              :product     product-id} api-key)
+        price-id (:id price)
+        subscription (subscriptions/create {:customer               customer-id
+                                            :default_payment_method (:id payment-method)
+                                            :items                  [{:price price-id}]} api-key)]
+    (testing "should be a valid subscription"
+      (is (some? (:id subscription))))))
```

Comment:
Is it worth testing the props here to ensure the mapping is working?

Reply:
True, I firstly wanted to get a basic version out to be able to run some integration tests, and then get back to the mapper and add more fields as required. I've added the assertions in anyway.